### PR TITLE
Added support for BlueROV2 Heavy

### LINKED ISSUE #8: Integrate BlueROV2 Heavy configuration model
### Problem Description

Blue Robotics offers an extension to the standard BlueROV2 called the Heavy configuration. This configuration is quite popular and enables full 6-DOF control (unlike the standard counterpart). It may be useful to offer an additional model for the BlueROV2 Heavy within this project.

### Proposal

Add a new `bluerov2-heavy` model to the models. This should include a visualization of the BlueROV2 Heavy using the [CAD model](https://grabcad.com/library/bluerobotics-bluerov2-heavy-1) and the identified hydrodynamic parameters previously determined from a system identification of the ROV (found [here](https://www.mdpi.com/2077-1312/10/12/1898)). Because the BuoyancyPlugin uses the collision volume to determine the ROV buoyancy, it will also be necessary to determine a suitable collision volume. One approach could be to use the mesh itself. Another alternative could be to extend the existing collision volume for the standard BlueROV2 (see [here](https://github.com/clydemcqueen/bluerov2_ignition/blob/9fc53aa13fe69a9a4376e1e8656519cd5e3644a3/models/bluerov2/model.sdf.in#L33)) to include a few additional rectangles encompassing the Heavy configuration wings.

### Alternative Solution

Rather than using the hydrodynamic parameters found from the system identification linked above, the same approach taken to calculate the parameters for the base BlueROV2 could be applied.

### PR DESCRIPTION
**Pull request by evan-palmer**
bluerov2_heavy

## Changes Made

This PR introduces the BlueROV2 Heavy model. Furthermore, a few small bugs were squashed, support for adding new models was improved, and tried to improve compliance with conventions established in the [SITL_Models](https://github.com/ArduPilot/SITL_Models/tree/master/Gazebo) project to make it easier to merge new models over there.

I remembered to squash my commits this time around too :wink: 

## Associated Issues

- Closes #8 

## Files Changed

- `README.md`: The README was updated to document usage for the new model.
- `models/bluerov2/**`: Modified some file names and added documentation to match the conventions of the [SITL_Models](https://github.com/ArduPilot/SITL_Models/tree/master/Gazebo) project, fixed a bug from the previous commit which used incorrect propeller meshes for some thrusters (didn't match the thruster configuration specified [here](https://www.ardusub.com/quick-start/vehicle-frame.html)), and improved the alignment of the thrusters that were previously unaligned.
- `models/bluerov2_heavy/**`: Introduced the BlueROV2 Heavy model which uses hydrodynamic parameters determined experiementally from previous work.
- `scripts/*.sh`: Updated the bash scripts to support different models
- `generate_model.py`: I had to do a fairly large re-write of this script to support generating different types of models. To make the tuning process a bit easier, I also added support for loading configurations from a YAML configuration file. I'd like to note that I considered placing all of the parameters into a dict directly instead of going from class -> dict, but I thought that a simple class might make it easier to add additional functionality in the future.
- `worlds/**`: Added a new world file for the heavy.

### CODE REVIEW

File: worlds/bluerov2_underwater.world [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" ?>
 <sdf version="1.6">
-  <world name="underwater">
+  <world name="bluerov2_underwater">
```
Comment (clydemcqueen) [inline worlds/bluerov2_underwater.world:3]: Hmmm... we could drop the _underwater suffix to make the world names a bit shorter.

File: scripts/down.sh [hunk 2]
```diff
@@ -1,3 +1,9 @@
 # Sends to cmd_thrust
-gz topic -t /model/bluerov2/joint/thruster5_joint/cmd_thrust -m gz.msgs.Double -p 'data: 5'
-gz topic -t /model/bluerov2/joint/thruster6_joint/cmd_thrust -m gz.msgs.Double -p 'data: 5'
+
+if [ $# -eq 0 ]; then
+    echo "Usage: $0 <model_name>"
+    exit 1
+fi
+
+gz topic -t /model/$1/joint/thruster5_joint/cmd_thrust -m gz.msgs.Double -p 'data: 5'
```
Comment (clydemcqueen) [inline scripts/down.sh:8]: This only spins 2 of the 4 vertical thrusters.

File: scripts/up.sh [hunk 3]
```diff
@@ -1,3 +1,9 @@
 # Sends to cmd_thrust
-gz topic -t /model/bluerov2/joint/thruster5_joint/cmd_thrust -m gz.msgs.Double -p 'data: -5'
-gz topic -t /model/bluerov2/joint/thruster6_joint/cmd_thrust -m gz.msgs.Double -p 'data: -5'
+
+if [ $# -eq 0 ]; then
+    echo "Usage: $0 <model_name>"
+    exit 1
+fi
+
+gz topic -t /model/$1/joint/thruster5_joint/cmd_thrust -m gz.msgs.Double -p 'data: -5'
```
Comment (clydemcqueen) [inline scripts/up.sh:8]: Ditto (only spins 2 thrusters)

File: scripts/generate_model.py [hunk 4]
```diff
@@ -1,170 +1,365 @@
...
-def thrust_to_ang_vel(thrust):
+# Density of water
+fluid_density = 1000.0
+
+
+def thrust_to_ang_vel(
+    thrust: float,
+    propeller_diameter: float,
+    thrust_coefficient: float,
+    fluid_density: float = 998,
```
Comment (clydemcqueen) [inline scripts/generate_model.py:65]: This shadows the global var of the same name, and the default is different. I like the config files; why not move fluid_density into the config? This gets rid of all globals except for the helper "dxxx" vars.

**Review by clydemcqueen — APPROVED** [review]
Wow, this is amazing! What a great contribution. Sorry it took so long to review this. A few nits.

File: worlds/bluerov2_underwater.world (hunk 1 above)
Comment (evan-palmer) [reply worlds/bluerov2_underwater.world:3]: Do you think that the filename should be updated as well? It looks like the [SITL_Models](https://github.com/ArduPilot/SITL_Models/tree/master/Gazebo/worlds) repo prefers to keep the environment descriptor in the filename.

Comment (clydemcqueen) [reply worlds/bluerov2_underwater.world:3]: Hmm... let's keep the same name.

**Comment by evan-palmer** [comment]
> Wow, this is amazing! What a great contribution. Sorry it took so long to review this. A few nits.

Thank you for the feedback! @ssicari was kind enough to make the changes that you have requested.

**Pull request merged**